File: tests/typesafety/python/money_pence.py
# expect: passes
# run: yes
# title: Integer pence add up exactly, and carry into shillings and pounds
# improves: money_float
def total_pence(prices_pence: list[int]) -> int:
    return sum(prices_pence)


def show(pence: int) -> str:
    shillings, d = divmod(pence, 12)
    pounds, s = divmod(shillings, 20)
    return f"£{pounds} {s}s {d}d"


assert total_pence([24, 24, 24]) == 72
assert show(72) == "£0 6s 0d"

File: tests/typesafety/typescript/money_pence.ts
// expect: passes

function totalPence(pricesPence: readonly number[]): number {
  return pricesPence.reduce((sum, price) => sum + price, 0);
}

function show(pence: number): string {
  const shillings = Math.floor(pence / 12);
  const d = pence % 12;
  return `£${Math.floor(shillings / 20)} ${shillings % 20}s ${d}d`;
}

export const sumIsExact = totalPence([24, 24, 24]) === 72;
export const shown = show(72);
